gpii/node_modules/gsettingsBridge/nodegsettings/nodegsettings.cc

```diff
@@ -15,119 +16,132 @@
+using v8::Local;
+using v8::Object;
+using v8::String;
+using Nan::Encode;
+using Nan::GetFunction;
+using Nan::New;
+using Nan::Set;
+using Nan::ThrowError;
 
-Handle<Value> get_gsetting_keys(const Arguments& args) {
-  HandleScope scope;
+NAN_METHOD(get_gsetting_keys) {
   GSettings *settings;
-  char schema[1024];
-  gchar ** keys;
+  gchar **keys;
   gint i;
   gint size = 0;
-  args[0]->ToString()->WriteAscii(schema);
-  settings = g_settings_new(schema);
+  Local<String> schema = info[0].As<String>();
+  settings = g_settings_new(*String::Utf8Value(schema));
   keys = g_settings_list_keys(settings);
+
   for (i = 0; keys[i]; i++) {
     size++; // Figure out how to do this in 1 loop
   }
-  Handle<Array> togo;
-  togo = Array::New(size);
+  Local<Array> togo = New<Array>(size);
   for (i = 0; keys[i]; i++) {
-    togo->Set(i,String::New(keys[i]));
+    togo->Set(i, Nan::Encode(keys[i], strlen(keys[i]), Nan::UTF8));
   }
-  return scope.Close(togo);
+
+  g_free(keys);
+  g_clear_object(&settings);
+
+  info.GetReturnValue().Set(togo);
 }
 
-/* Should take schema and key */
-Handle<Value> get_gsetting(const Arguments& args) {
-  HandleScope scope;
+///* Should take schema and key */
+NAN_METHOD(get_gsetting) {
   GSettings *settings;
-  char schema[1024];
-  char key[1024];
-  args[0]->ToString()->WriteAscii(schema);
-  settings = g_settings_new(schema);
-  args[1]->ToString()->WriteAscii(key);
+  Local<String> schema = info[0].As<String>();
+  Local<String> key = info[1].As<String>();
+  settings = g_settings_new(*String::Utf8Value(schema));
   GVariant* variant;
   const GVariantType* type;
 
-  variant = g_settings_get_value(settings,key);
+  variant = g_settings_get_value(settings, *String::Utf8Value(key));
   type = g_variant_get_type(variant);
 
   if (g_variant_type_equal(type,G_VARIANT_TYPE_DOUBLE)) {
-    return scope.Close(Number::New(g_variant_get_double(variant)));
+    info.GetReturnValue().Set(New(g_variant_get_double(variant)));
   }
   else if (g_variant_type_equal(type,G_VARIANT_TYPE_INT32)) {
-    return scope.Close(Int32::New(g_variant_get_int32(variant)));
+    info.GetReturnValue().Set(New(g_variant_get_int32(variant)));
   }
   else if (g_variant_type_equal(type,G_VARIANT_TYPE_STRING)) {
-    return scope.Close(String::New(g_variant_get_string(variant,NULL)));
+    info.GetReturnValue().Set(New(g_variant_get_string(variant,NULL)).ToLocalChecked());
   }
   else if (g_variant_type_equal(type,G_VARIANT_TYPE_BOOLEAN)) {
-    return scope.Close(Boolean::New(g_variant_get_boolean(variant)));
+    info.GetReturnValue().Set(New<Boolean>(g_variant_get_boolean(variant)));
   }
   else {
     g_print("The type is %s\n", g_variant_type_peek_string(type));
-    ThrowException(Exception::Error(String::New("Need to implement reading that value type")));
-    return scope.Close(Undefined());
+    ThrowError(New("Need to implement reading that value type").ToLocalChecked());
+    info.GetReturnValue().Set(Nan::False());
```

Comment:
This line will be unreachable. If our cleanup hasn't been properly set up by now (i.e. via a destructor) it will be too late.